client/allocrunner/alloc_runner_test.go

```diff
@@ -1811,3 +1836,90 @@ func TestAllocRunner_Lifecycle_Shutdown_Order(t *testing.T) {
 	last = upd.Last()
 	require.Less(t, last.TaskStates[sidecarTask.Name].FinishedAt, last.TaskStates[poststopTask.Name].FinishedAt)
 }
+
+func TestHasSidecarTasks(t *testing.T) {
+	ci.Parallel(t)
+
+	falseV, trueV := false, true
+
+	cases := []struct {
+		name string
+		// nil if main task, false if non-sidecar hook, true if sidecar hook
+		indicators []*bool
```

Comment:
Seems like we could make this more obvious by using a little 3 state enum

Reply:
Ah good point. I just copied this test from another file and didn't pay much attention to it 😅